From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Tue, 18 Jul 2023 22:42:43 +0200
Subject: [PATCH] Client perspective - For players - Set based on resource pack
 status

License: Fiddle Public License Agreement 1.0 (included in license/FPL-1.0.txt)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 820836fda03ab2fe970b3f8881e0d75d4c817b45..fac99a6fec1c172fd6a75a625a0fd2d660cd97b5 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -101,6 +101,8 @@ import net.minecraft.world.scores.Objective;
 import net.minecraft.world.scores.PlayerTeam;
 import net.minecraft.world.scores.Scoreboard; // Paper
 import net.minecraft.world.scores.Team;
+import org.bukkit.event.player.*;
+import org.fiddlemc.fiddle.packet.ClientPerspectiveCategory;
 import org.slf4j.Logger;
 
 // CraftBukkit start
@@ -121,13 +123,7 @@ import org.bukkit.craftbukkit.entity.CraftPlayer;
 import org.bukkit.craftbukkit.util.CraftChatMessage;
 import org.bukkit.craftbukkit.util.CraftLocation;
 import org.bukkit.entity.Player;
-import org.bukkit.event.player.PlayerChangedWorldEvent;
-import org.bukkit.event.player.PlayerJoinEvent;
-import org.bukkit.event.player.PlayerLoginEvent;
-import org.bukkit.event.player.PlayerQuitEvent;
-import org.bukkit.event.player.PlayerRespawnEvent;
 import org.bukkit.event.player.PlayerRespawnEvent.RespawnReason;
-import org.bukkit.event.player.PlayerSpawnChangeEvent;
 // CraftBukkit end
 
 public abstract class PlayerList {
@@ -211,6 +207,7 @@ public abstract class PlayerList {
         player.hasBeenPlacedIntoPlayerList = true;
         Connection connection = playerconnection.connection;
         // Fiddle end - client perspective - send resource pack
+        player.setClientPerspective(player.getBukkitEntity().getResourcePackStatus() == PlayerResourcePackStatusEvent.Status.SUCCESSFULLY_LOADED ? ClientPerspectiveCategory.JAVA_WITH_ENABLED_RESOURCE_PACK.mostLikelyPerspective : ClientPerspectiveCategory.JAVA_DEFAULT.mostLikelyPerspective); // Fiddle - client perspective - player - set based on resource pack status
         player.isRealPlayer = true; // Paper
         player.loginTime = System.currentTimeMillis(); // Paper
         GameProfile gameprofile = player.getGameProfile();
